docs/Custom_Operation_for_GPUs.md

```diff
@@ -0,0 +1,2323 @@
+# Custom operation for GPUs
+
+JAX, a popular framework for machine learning, comes with a large number of operations, however, users occasionally run into a situation where they need a new operation that is not supported by JAX.
+To accommodate such scenarios, JAX allows users to define custom operations and this tutorial is to explain how we can define one for GPUs and use it in single-GPU and multi-GPU environments.
+
+This tutorial contains information from [Extending JAX with custom C++ and CUDA code](https://github.com/dfm/extending-jax).
+
+# RMS Normalization
+
+For this tutorial, we are going to add the RMS normalization as a custom operation in JAX.
```

Comment:
It's worth noting that RMS norm is quite easy to express in jax.numpy directly, but i guess your point is that a custom CUDA kernel could be more performant?

Reply:
Added an explanation